api/core/v1alpha1/clustercatalog_types.go

```diff
@@ -85,153 +115,269 @@ type ClusterCatalogSpec struct {
 	//    image:
 	//      ref: quay.io/operatorhubio/catalog:latest
 	//
-	// For more information on FBC, see https://olm.operatorframework.io/docs/reference/file-based-catalogs/#docs
+	// +kubebuilder:validation:Required
 	Source CatalogSource `json:"source"`
 
-	// priority is an optional field that allows the user to define a priority for a ClusterCatalog.
+	// priority allows the user to define a priority for a ClusterCatalog.
+	// priority is optional.
+	//
 	// A ClusterCatalog's priority is used by clients as a tie-breaker between ClusterCatalogs that meet the client's requirements.
+	// It is up to clients to decide how to handle scenarios where multiple ClusterCatalogs with the same priority meet their requirements.
 	// For example, in the case where multiple ClusterCatalogs provide the same bundle.
-	// A higher number means higher priority. Negative numbers are also accepted.
-	// When omitted, the default priority is 0.
+	// A higher number means higher priority.
+	//
+	// When omitted, the default priority is 0 because that is the zero value of integers.
+	//
+	// Negative numbers can be used to specify a priority lower than the default.
+	// Positive numbers can be used to specify a priority higher than the default.
+	//
+	// The lowest possible value is -2147483648.
+	// The highest possible value is 214748647.
```

Comment:
Is this where we landed? I know Joel was concerned about the use case of "how to I set the maximum or minimum priority?" And if users want to do that, they'll have to pick this pretty random looking set of numbers. What if we round down like this:
```suggestion
	// The lowest possible value is -2000000000.
	// The highest possible value is 2000000000.
```

Reply:
Would it help the "randomness" aspect if we mentioned specifically that this is a 32-bit integer value? I'm not strongly opinionated on the value here, but if the concern is about this appearing random we can be a bit more explicit as to why these are the minimum and maximum values

Reply:
:shrug:

I found the original thread on this subject and forgot that another reason to leave it like this is because OLMv0 has this scheme with these same limits, and the motivation would be to make migrations easier.

But having thought about that aspect more, it would be trivial to preflight a "CatalogSource to ClusterCatalog" migration by looking at the priority fields of all of the CatalogSources and failing out early if any of them have values outside the new range.  This would likely hit very few clusters (only those that define custom CatalogSources, and of those, only ones that fall at the extreme ends of the values).

Reply:
I don't this "this looks kind of random" is a strong enough reason for me to be on board with tightening the range here and forcing our migration story to have to handle this scenario.

In lieu of a stronger motivation, I'm going to leave this as is.